consensus/src/consensus/mod.rs

```diff
@@ -318,6 +319,8 @@ impl Consensus {
         let (virtual_sender, virtual_receiver): (CrossbeamSender<BlockProcessingMessage>, CrossbeamReceiver<BlockProcessingMessage>) =
             unbounded_crossbeam();
 
+        let notification_root = Arc::new(ConsensusNotificationRoot::new(notification_sender));
```

Comment:
Maybe pass the root to consensus and not the sender? this way you can avoid the `notification_root(&self)` getter. I think it will play out better when we have a staging consensus and we need to switch the active consensus to it

Reply:
You can keep `TestConsensus` as is, so you will not need to change all tests